common/lib/xmodule/xmodule/contentstore/mongo.py

```diff
@@ -102,11 +102,14 @@ def save(self, content):
 
             # It seems that this code thought that only some specific object would have the `__iter__` attribute
             # but the bytes object in python 3 has one and should not use the chunking logic.
-            if hasattr(content.data, '__iter__') and not isinstance(content.data, six.binary_type):
+            if hasattr(content.data, '__iter__') and not isinstance(content.data, (six.binary_type,six.string_types)):
                 for chunk in content.data:
                     fp.write(chunk)
             else:
-                fp.write(content.data)
+                if isinstance(content.data, six.text_type):
```

Comment:
Why is it necessary to allow for 2 different types for `content.data`?  Would it make sense to `assert` that the type is of one expected value so we can fix any callers that aren't abiding by the interface?

Reply:
Because this is deep in contentstore gridfs storage, it's accessed from a lot of places that I don't know if I can condifently enumerate and fix.  Gridfs' interface is to write bytes data so I'm ensuring that's the case no matter what.  If you feel strongly about making this change further upstream, I can also add a warning that logs the parent caller and we could come back and fix this later.

Reply:
Ok. So we're taking a safe approach for now - which makes sense in this initial rollout.  Can you add a comment (either in this PR or in a follow-up PR) to this effect so it's clearer why we chose this route?

Reply:
Will do!